packages/pluggableWidgets/video-player-native/src/VideoPlayer.tsx

```diff
@@ -36,27 +53,129 @@ export function VideoPlayer(props: VideoPlayerProps<VideoStyle>): ReactElement {
-                    setVideoAspectRatio(data.naturalSize.width / data.naturalSize.height);
-                }}
-                onError={() => setStatus(StatusEnum.ERROR)}
-                style={status !== StatusEnum.READY ? { height: 0 } : styles.video}
-                useTextureView={false}
-                resizeMode={props.aspectRatio ? "contain" : "stretch"}
-            />
-        </View>
+        <Fragment>
+            {isAndroid && (
+                <Modal
+                    isVisible={fullScreen}
+                    style={styles.fullScreenVideoPlayer}
+                    onBackButtonPress={() => fullScreenHandler(false)}
+                    testID="fullscreen-modal"
+                >
+                    <TouchableWithoutFeedback onPress={onVideoPressHandler} testID="fullscreen-overlay">
+                        <Video
+                            {...videoProps}
+                            ref={fullScreenPlayerRef}
+                            paused={false}
+                            onLoad={data => {
+                                setStatus(StatusEnum.READY);
+                                setVideoAspectRatio(data.naturalSize.width / data.naturalSize.height);
+                                fullScreenPlayerRef.current?.seek(currentPlayTime);
+                            }}
+                            testID={`${props.name}-fullscreen`}
+                            style={styles.fullScreenVideoStyle}
+                        />
+                    </TouchableWithoutFeedback>
+                    {showControls && (
+                        <TouchableOpacity
+                            style={styles.controlBtnContainerStyle}
+                            onPress={() => {
+                                fullScreenHandler(false);
+                            }}
+                            testID="btn-fullscreen-exit"
+                        >
+                            <Icon name="fullscreen-exit" color="white" size={22} />
+                        </TouchableOpacity>
+                    )}
+                    {status === StatusEnum.LOADING && (
+                        <ActivityIndicator
+                            color={styles.indicator.color}
+                            size="large"
+                            style={styles.fullScreenActivityIndicatorStyle}
+                        />
+                    )}
+                    {errorText}
+                </Modal>
+            )}
+            <View style={styles.container}>
+                {status === StatusEnum.LOADING && <ActivityIndicator color={styles.indicator.color} size="large" />}
+                {errorText}
+                {!fullScreen && (
+                    <TouchableWithoutFeedback style={styles.container} onPress={onVideoPressHandler}>
+                        <Video
+                            {...videoProps}
+                            paused={!props.autoStart && !currentPlayTime}
+                            onLoad={data => {
```

Comment:
I suggets using `useCallback`

Reply:
On second thought this may not be helpful. Best to check. Let me know what you think

Reply:
same as [here](https://github.com/mendix/widgets-resources/pull/1303/files#r807777014)